### DISCUSSION TITLE
`resting_stratified_bumpy_ocean.jl` not running

### DISCUSSION THREAD

**Discussion by francispoulin** [CliMA/Oceananigans.jl#2992] [General]
I tried running the above in validation and it no longer works.  I get the following error.  

Can someone suggest how this should be updated?  I am happy to try it out.

```
ERROR: LoadError: ArgumentError: The dimensions of the immersed boundary (1, 134, 1) do not match the grid size (1, 134)
```

**Comment by glwagner** [comment]
Link to file?

**Reply by francispoulin** [reply]
Sorry for not doing this earlier but here goes.

https://github.com/CliMA/Oceananigans.jl/blob/main/validation/immersed_boundaries/resting_stratified_bumpy_ocean.jl

**Reply by glwagner** [reply]
We can use functions with immersed boundaries now, so this might work:

```julia
immersed_boundaries = [
                       PartialCellBottom(seamount;
                                         minimum_fractional_Δz),
                       GridFittedBottom(seamount)
                      ]
````

and you don't need `seamount_field`.

**Reply by glwagner** [reply]
This would probably work too:

```julia
immersed_boundaries = [
                       PartialCellBottom(interior(seamount_field, :, :, 1);
                                         minimum_fractional_Δz),
                       GridFittedBottom(interior(seamount_field, :, :, 1))
                      ]

**Reply by francispoulin** [reply]
Thanks!

I can confirm that the first one works.  Since this method doesn't require defining `seamount_field`, that seems the better way to go.  

I'll create a PR right now with this update.

**Reply by glwagner** [reply]
great, thanks

**Comment by francispoulin** [comment]
One more question @glwagner on this.

I noticed that if I specified the advection schemes to be WENO, then an error occurs because the immersed boundary is (1, 136) and it does not match the grid size (1, 138).  I tried increasin the halo to 4 in all directions and that didn't help.

I am pretty sure I am doing something wrong but I don't know what am I missing here.  Any suggestions?

```
tracer_advection = WENO(grid = grid),
momentum_advection = WENO(grid = grid),
```

**Reply by glwagner** [reply]
This is the bug in #2988 ... try

```julia
tracer_advection = WENO(grid.underlying_grid)
momentum_advection = WENO(grid.underlying_grid)
```

@simone-silvestri

**Reply by glwagner** [reply]
sorry about that, it's very annoying.

**Reply by francispoulin** [reply]
Thanks for the suggestion and will try it out soon.  But now to catch a plane Boston. Too bad I won't get to visit MIT as I go straight to WHOI for a workshop.
